bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/codemining/CodeMiningManager.java

```diff
@@ -194,7 +195,7 @@ private static CompletableFuture<List<? extends ICodeMining>> getCodeMinings(ITe
 				}))
 				.collect(Collectors.toList());
 		return CompletableFuture.allOf(com.toArray(new CompletableFuture[com.size()])).thenApply(
-				v -> com.stream().map(CompletableFuture::join).flatMap(java.util.Collection::stream).collect(Collectors.toList()));
+				v -> com.stream().map(CompletableFuture::join).filter(Objects::nonNull).flatMap(java.util.Collection::stream).collect(Collectors.toList()));
```

Comment:
Could you please explain why it breaks the contract? Is it an intended feature that one provider can stop rendering of all minings if null is returned in the CompletableFutre? For me, it is a bug.

Reply:
My reasoning was that now there is one provider which is missing in the output. But since it is null, we don't expect much to come from it, so I now agree that this helps in better fulfilling the contract.
Sorry for the confusion!